packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts

```diff
@@ -36,13 +36,8 @@ export const useGridFocus = (
 
   const setCellFocus = React.useCallback<GridFocusApi['setCellFocus']>(
     (id, field) => {
-      // The row might have been deleted
-      if (!apiRef.current.getRow(id)) {
-        return;
-      }
-
       const focusedCell = gridFocusCellSelector(apiRef);
-      if (focusedCell?.id === id && focusedCell.field === field) {
+      if (focusedCell?.id === id && focusedCell?.field === field) {
```

Comment:
Without "?", https://deploy-preview-4831--material-ui-x.netlify.app/x/react-data-grid/scrolling/#scrolling-to-specific-cells crashes.